File: Data Structures & Algorithms/reverse-integer/submission-0.cpp
class Solution {
public:
    int reverse(int x) {
        int res=0;
        bool neg = x<0;
        x = neg? -x : x;
        while(x!=0) {
            if(res>(INT_MAX-x%10)/10) return 0;
            res = res*10 + x%10;
            x /= 10;
        }
        return neg?-res:res;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root: return []
        prev=[root]
        nxt=[]
        res=[]

        while prev:
            res.append(prev[-1].val)
            for node in prev:
                if node.left: nxt.append(node.left)
                if node.right: nxt.append(node.right)

            prev=nxt
            nxt=[]

        return res